### ISSUE TITLE
Change default renderType to createRoot in React 18

### ISSUE THREAD

**Issue by joeldenning** [single-spa/single-spa-react#115]
React 18 console warns when you run ReactDOM.render() rather than ReactDOM.createRoot().render(). I think single-spa-react should do a React version check and start defaulting to createRoot if it's at least react 18.

**Comment by joeldenning** [comment]
Here are the lines to change - should check whether it's react 18 or not when determining the default:

https://github.com/single-spa/single-spa-react/blob/025245b5b00b56991c38a70dd2ab189f7134b8d0/src/single-spa-react.js#L269-L270

**Comment by frehner** [comment]
You think a version check is what you want, or do you want to do an existence check?

```js
if(ReactDOM.createRoot) {
  ... new stuff
} else {
  ... old stuff
}
```

It appears that `createRoot` isn't defined on ReactDOM in v17, so I think it would be safe?

**Comment by joeldenning** [comment]
Yeah probably just checking whether createRoot exists is enough. Experimental builds of react 16 and 17 had it, but those using the experimental builds probably would want to use createRoot since that was the whole purpose of using those builds.

**Comment by filoxo** [comment]
Updated React 18 createRoot information here: https://github.com/reactwg/react-18/discussions/5

**Comment by hijiangtao** [comment]
It seems that `ReactDOM: typeof ReactDOM;` (which is not correct for React v18+ users) should be updated from `SingleSpaReactOpts`, since React v18 has already be supported in the latest version, otherwise, it will raise a TypeScript incompatible error with following message:

```
TS2740: Type 'typeof import("/Users/hijiangtao/Documents/git/weekly-management/node_modules/@types/react-dom/client")' is missing the following properties from type 'typeof import("/Users/hijiangtao/Documents/git/weekly-management/node_modules/@types/react-dom/index")': findDOMNode, unmountComponentAtNode, createPortal, flushSync, and 5 more.  

single-spa-react.d.ts(9, 3): The expected type comes from property 'ReactDOM' which is declared here on type 'SingleSpaReactOpts<AppProps>'
```

**Comment by filoxo** [comment]
Published in v5.0.0

**Comment by serpilacar** [comment]
@filoxo [In docs ](https://single-spa.js.org/docs/ecosystem-react/#options)it still declared as `render` is the default renderType. I thought this PR referred in this issue changed the default renderType to `createRoot` 🤔

**Comment by italoservio** [comment]
> Published in v5.0.0

**Link**
https://github.com/single-spa/single-spa-react/releases/tag/v5.0.0

**What's Changed**
_React 18 support. by @filoxo in_ https://github.com/single-spa/single-spa-react/pull/148
- if upgrading from single-spa-react@4, replace ReactDOM with ReactDOMClient Change default renderType to createRoot. by @filoxo in https://github.com/single-spa/single-spa-react/pull/148

- if your application is using React 17 and below, you must explicitly define the renderType option. For most, this is "render" but [others are also supported](https://github.com/single-spa/single-spa-react/blob/main/types/single-spa-react.d.ts#L6-L11).
